### PR TITLE
feat: remove synth init

### LINKED ISSUE #83: UX: Make synth init obsolete
**Required Functionality**
Currently, users must initialize a "workspace" before using synth to generate data. The rationale for this was that synth would be able to write files for both import and possibly generation (e.g. with #33) and having a directory for itself would reduce the risk of accidentally overwriting files. However, both requiring a call to "synth init" and the generated directories, even if empty, make it harder to set up tests and worsen the user experience.

**Proposed Solution**
Recognizing that we put the baby out with the bathwater, we should not require user interaction unless there's a problem. So at the very least we could remove the synth init command and subsequent check for the `.synth` directory and instead add a prompt to ask the user whenever overwriting a file. Even better, we could record the files and creation timestamp whenever synth creates a new file and not issue a prompt when a file of that list gets overwritten. We should however keep the list of generated files somewhat short; storing up to 1000 files should be enough for everybody (famous last words). When the list gets full, we could check for removed files, otherwise remove the oldest entries.  

**Use case**
Improve user experience, simplify testing and reproducibility (because the tests no longer need to clean up `.synth` folders).

### PR DESCRIPTION
**Pull request by brokad**
This removes `synth init` and references to it in documentation, in CI and README.

Closes #83.

### CODE REVIEW

**Comment by brokad** [comment]
@christoshadjiaslanis I have added a no-op `synth init` command. Can you take a look and tell me if that's OK?

File: synth/src/cli/mod.rs [hunk 1]
```diff
@@ -270,19 +195,18 @@ impl Cli {
 #[derive(StructOpt)]
 #[structopt(name = "synth", about = "synthetic data engine on the command line")]
 pub enum Args {
-    #[structopt(about = "Initialise the workspace")]
+    #[structopt(about = "(DEPRECATED). Initialize a new workspace")]
```
Comment (christoshadjiaslanis) [inline synth/src/cli/mod.rs:198]: Perhaps explain that it is now a no-op?

**Comment by christoshadjiaslanis** [comment]
Left one comment - other than that LGTM.

File: synth/src/cli/mod.rs (hunk 1 above)
Comment (brokad) [reply synth/src/cli/mod.rs]: Thank you!

**Pull request merged**